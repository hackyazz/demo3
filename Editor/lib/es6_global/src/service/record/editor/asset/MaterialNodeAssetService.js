

import * as Block from "../../../../../../../node_modules/bs-platform/lib/es6/block.js";
import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Log$WonderLog from "../../../../../../../node_modules/wonder-log/lib/es6_global/src/Log.js";
import * as LogUtils$WonderEditor from "../../../../core/utils/console/LogUtils.js";

function getNodeName(param, getMaterialNameFunc) {
  return Curry._2(getMaterialNameFunc, param[/* materialComponent */1], param[/* type_ */0]);
}

function buildNodeData(type_, materialComponent) {
  return /* record */[
          /* type_ */type_,
          /* materialComponent */materialComponent
        ];
}

function buildNode(nodeId, type_, materialComponent) {
  return /* MaterialNode */Block.__(1, [
            nodeId,
            /* record */[
              /* type_ */type_,
              /* materialComponent */materialComponent
            ]
          ]);
}

function buildNodeByNodeData(nodeId, nodeData) {
  return /* MaterialNode */Block.__(1, [
            nodeId,
            nodeData
          ]);
}

function isMaterialNode(node) {
  if (node.tag === 1) {
    return true;
  } else {
    return false;
  }
}

function getNodeData(node) {
  if (node.tag === 1) {
    return node[1];
  } else {
    return Log$WonderLog.fatal(LogUtils$WonderEditor.buildFatalMessage("should be material node", "", "", ""));
  }
}

export {
  getNodeName ,
  buildNodeData ,
  buildNode ,
  buildNodeByNodeData ,
  isMaterialNode ,
  getNodeData ,
  
}
/* Log-WonderLog Not a pure module */
