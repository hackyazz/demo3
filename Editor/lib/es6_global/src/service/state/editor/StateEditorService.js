

import * as CreateEditorStateDataEditorService$WonderEditor from "../data/CreateEditorStateDataEditorService.js";

function getStateIsDebug(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1];
}

function setStateIsDebug(isDebug) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1] = isDebug;
  return /* () */0;
}

function getIsRun(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2];
}

function setIsRun(isRun) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2] = isRun;
  return /* () */0;
}

function getState(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0];
}

function setState(state) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0] = state;
  return state;
}

export {
  getStateIsDebug ,
  setStateIsDebug ,
  getIsRun ,
  setIsRun ,
  getState ,
  setState ,
  
}
/* CreateEditorStateDataEditorService-WonderEditor Not a pure module */
