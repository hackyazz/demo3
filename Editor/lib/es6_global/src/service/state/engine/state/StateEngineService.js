

import * as StateAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/StateAPI.js";

var deepCopyForRestore = StateAPI$Wonderjs.deepCopyForRestore;

var createState = StateAPI$Wonderjs.createState;

var createStateData = StateAPI$Wonderjs.createStateData;

var getStateFromData = StateAPI$Wonderjs.getStateFromData;

var unsafeGetState = StateAPI$Wonderjs.unsafeGetState;

var setStateToData = StateAPI$Wonderjs.setStateToData;

var setState = StateAPI$Wonderjs.setState;

var setIsDebug = StateAPI$Wonderjs.setIsDebug;

export {
  deepCopyForRestore ,
  createState ,
  createStateData ,
  getStateFromData ,
  unsafeGetState ,
  setStateToData ,
  setState ,
  setIsDebug ,
  
}
/* StateAPI-Wonderjs Not a pure module */
