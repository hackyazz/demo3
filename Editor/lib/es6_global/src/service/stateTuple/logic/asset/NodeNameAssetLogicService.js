

import * as NodeAssetService$WonderEditor from "../../../record/editor/asset/NodeAssetService.js";
import * as WDBNodeAssetService$WonderEditor from "../../../record/editor/asset/WDBNodeAssetService.js";
import * as NodeNameAssetService$WonderEditor from "../../../record/editor/asset/NodeNameAssetService.js";
import * as FolderNodeAssetService$WonderEditor from "../../../record/editor/asset/FolderNodeAssetService.js";
import * as OperateTextureLogicService$WonderEditor from "../OperateTextureLogicService.js";
import * as OperateMaterialLogicService$WonderEditor from "../material/OperateMaterialLogicService.js";

function getFolderNodeName(node) {
  return FolderNodeAssetService$WonderEditor.getNodeName(FolderNodeAssetService$WonderEditor.getNodeData(node));
}

function getWDBNodeName(node) {
  return WDBNodeAssetService$WonderEditor.getNodeName(WDBNodeAssetService$WonderEditor.getNodeData(node));
}

var getTextureNodeName = OperateTextureLogicService$WonderEditor.getName;

var getMaterialNodeName = OperateMaterialLogicService$WonderEditor.getName;

function getNodeName(node, engineState) {
  return NodeNameAssetService$WonderEditor.getNodeName(node, (function (param) {
                return OperateTextureLogicService$WonderEditor.getName(param, engineState);
              }), (function (param) {
                return (function (param$1) {
                    return OperateMaterialLogicService$WonderEditor.getName(param, param$1, engineState);
                  });
              }));
}

function isNodeEqualByName(sourceNode, targetNode, engineState) {
  return NodeNameAssetService$WonderEditor.isNodeEqualByName(sourceNode, targetNode, (function (param) {
                return OperateTextureLogicService$WonderEditor.getName(param, engineState);
              }), (function (param) {
                return (function (param$1) {
                    return OperateMaterialLogicService$WonderEditor.getName(param, param$1, engineState);
                  });
              }));
}

function isTargetNameNode(node, name, engineState) {
  return NodeNameAssetService$WonderEditor.isTargetNameNode(node, name, (function (param) {
                return OperateTextureLogicService$WonderEditor.getName(param, engineState);
              }), (function (param) {
                return (function (param$1) {
                    return OperateMaterialLogicService$WonderEditor.getName(param, param$1, engineState);
                  });
              }));
}

function updateNodeName(node, name, engineState) {
  return NodeAssetService$WonderEditor.handleNode(node, (function (nodeId, param) {
                return /* tuple */[
                        OperateTextureLogicService$WonderEditor.setName(param[/* textureComponent */0], name, engineState),
                        node
                      ];
              }), (function (nodeId, param) {
                return /* tuple */[
                        OperateMaterialLogicService$WonderEditor.setName(param[/* materialComponent */1], param[/* type_ */0], name, engineState),
                        node
                      ];
              }), (function (nodeId, nodeData) {
                return /* tuple */[
                        engineState,
                        WDBNodeAssetService$WonderEditor.buildNodeByNodeData(nodeId, WDBNodeAssetService$WonderEditor.rename(name, nodeData))
                      ];
              }), (function (nodeId, nodeData, children) {
                return /* tuple */[
                        engineState,
                        FolderNodeAssetService$WonderEditor.buildNodeByNodeData(nodeId, FolderNodeAssetService$WonderEditor.rename(name, nodeData), children)
                      ];
              }));
}

export {
  getFolderNodeName ,
  getWDBNodeName ,
  getTextureNodeName ,
  getMaterialNodeName ,
  getNodeName ,
  isNodeEqualByName ,
  isTargetNameNode ,
  updateNodeName ,
  
}
/* WDBNodeAssetService-WonderEditor Not a pure module */
