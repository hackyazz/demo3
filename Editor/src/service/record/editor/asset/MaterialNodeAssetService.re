open TreeAssetType;

open NodeAssetType;

let getNodeName =
    ({materialComponent, type_}: materialNodeData, getMaterialNameFunc) =>
  getMaterialNameFunc(~material=materialComponent, ~type_);

let buildNodeData = (~type_, ~materialComponent) => {
  type_,
  materialComponent,
};

let buildNode = (~nodeId, ~type_, ~materialComponent) =>
  MaterialNode(nodeId, buildNodeData(~type_, ~materialComponent));

let buildNodeByNodeData = (~nodeId, ~nodeData) =>
  MaterialNode(nodeId, nodeData);

let isMaterialNode = node =>
  switch (node) {
  | MaterialNode(_, _) => true
  | _ => false
  };

let getNodeData = node =>
  switch (node) {
  | MaterialNode(nodeId, nodeData) => nodeData
  | _ =>
    WonderLog.Log.fatal(
      LogUtils.buildFatalMessage(
        ~description={j|should be material node|j},
        ~reason="",
        ~solution={j||j},
        ~params={j||j},
      ),
    )
  };