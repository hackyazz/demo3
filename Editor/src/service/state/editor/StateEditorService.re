open EditorStateDataType;

let getStateIsDebug = () =>
  CreateEditorStateDataEditorService.editorStateData.isDebug;

let setStateIsDebug = isDebug => {
  CreateEditorStateDataEditorService.editorStateData.isDebug = isDebug;

  ();
};

let getIsRun = () => CreateEditorStateDataEditorService.editorStateData.isRun;

let setIsRun = isRun => {
  CreateEditorStateDataEditorService.editorStateData.isRun = isRun;

  ();
};

let getState = () =>
  CreateEditorStateDataEditorService.editorStateData.editorState;

let setState = state => {
  CreateEditorStateDataEditorService.editorStateData.editorState = state;
  state;
};