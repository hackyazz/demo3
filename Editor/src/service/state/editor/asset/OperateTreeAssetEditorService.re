open NodeAssetType;

let _getTree = (func, editorState) =>
  TreeAssetEditorService.unsafeGetTree(editorState) |> func;

let _getTreeWithLabel = (func, editorState) =>
  func(~tree=TreeAssetEditorService.unsafeGetTree(editorState));

let _getAndSetTree = (func, editorState) =>
  TreeAssetEditorService.unsafeGetTree(editorState)
  |> func
  |> TreeAssetEditorService.setTree(_, editorState);

let findNodeById = (targetNodeId, editorState) =>
  _getTree(OperateTreeAssetService.findNodeById(targetNodeId), editorState);

let unsafeFindNodeById = (targetNodeId, editorState) =>
  _getTree(
    OperateTreeAssetService.unsafeFindNodeById(targetNodeId),
    editorState,
  );

let isNodeExistById = (targetNodeId, editorState) =>
  findNodeById(targetNodeId, editorState) |> Js.Option.isSome;

let findMaterialNode = (targetMaterialComponent, targetType_, editorState) => {
  let predNodeFunc = materialNode => {
    let {materialComponent, type_} =
      MaterialNodeAssetService.getNodeData(materialNode);

    materialComponent === targetMaterialComponent && type_ === targetType_;
  };

  editorState
  |> _getTreeWithLabel(
       IterateTreeAssetService.findOne(
         ~predMaterialNodeFunc=predNodeFunc,
         (),
       ),
     );
};

let insertNode = (targetNodeId, newTreeNode, editorState) =>
  _getAndSetTree(
    OperateTreeAssetService.insertNode(targetNodeId, newTreeNode),
    editorState,
  );

let setNodeIsShowChildren = (targetNodeId, isShowChildren, editorState) =>
  _getAndSetTree(
    OperateTreeAssetService.setNodeIsShowChildren(
      targetNodeId,
      isShowChildren,
    ),
    editorState,
  );

let removeNode = (targetNode, editorState) => {
  let newTree =
    TreeAssetEditorService.unsafeGetTree(editorState)
    |> OperateTreeAssetService.removeNode(targetNode);

  TreeAssetEditorService.setTree(newTree, editorState);
};

let removeNodeById = (targetNodeId, editorState) => {
  let (newTree, removedNode) =
    TreeAssetEditorService.unsafeGetTree(editorState)
    |> OperateTreeAssetService.removeNodeById(targetNodeId);

  (newTree |> TreeAssetEditorService.setTree, removedNode);
};

let findNodeParent = (targetNode, editorState) =>
  IterateTreeAssetService.findOne(
    ~tree=TreeAssetEditorService.unsafeGetTree(editorState),
    ~predFolderNodeFunc=
      node =>
        FolderNodeAssetService.findChild(node, targetNode) |> Js.Option.isSome,
    (),
  );

let findNodeParentId = (targetNode, editorState) =>
  findNodeParent(targetNode, editorState)
  |> Js.Option.map((. node) => NodeAssetService.getNodeId(~node));

let getCurrentNode = editorState =>
  CurrentNodeIdAssetEditorService.getCurrentNodeId(editorState)
  |> Js.Option.map((. nodeId) => unsafeFindNodeById(nodeId, editorState));

let unsafeGetSelectedFolderNodeInAssetTree = editorState =>
  TreeAssetEditorService.getSelectedFolderNodeIdInAssetTree(editorState)
  |> unsafeFindNodeById(_, editorState);