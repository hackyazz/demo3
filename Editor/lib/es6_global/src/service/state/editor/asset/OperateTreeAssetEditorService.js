

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Js_option from "../../../../../../../node_modules/bs-platform/lib/es6/js_option.js";
import * as NodeAssetService$WonderEditor from "../../../record/editor/asset/NodeAssetService.js";
import * as FolderNodeAssetService$WonderEditor from "../../../record/editor/asset/FolderNodeAssetService.js";
import * as TreeAssetEditorService$WonderEditor from "./TreeAssetEditorService.js";
import * as IterateTreeAssetService$WonderEditor from "../../../record/editor/asset/IterateTreeAssetService.js";
import * as OperateTreeAssetService$WonderEditor from "../../../record/editor/asset/OperateTreeAssetService.js";
import * as MaterialNodeAssetService$WonderEditor from "../../../record/editor/asset/MaterialNodeAssetService.js";
import * as CurrentNodeIdAssetEditorService$WonderEditor from "./CurrentNodeIdAssetEditorService.js";

function _getTree(func, editorState) {
  return Curry._1(func, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
}

function _getTreeWithLabel(func, editorState) {
  return Curry._1(func, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
}

function _getAndSetTree(func, editorState) {
  var __x = Curry._1(func, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
  return TreeAssetEditorService$WonderEditor.setTree(__x, editorState);
}

function findNodeById(targetNodeId, editorState) {
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return OperateTreeAssetService$WonderEditor.findNodeById(targetNodeId, param);
}

function unsafeFindNodeById(targetNodeId, editorState) {
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return OperateTreeAssetService$WonderEditor.unsafeFindNodeById(targetNodeId, param);
}

function isNodeExistById(targetNodeId, editorState) {
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return Js_option.isSome(OperateTreeAssetService$WonderEditor.findNodeById(targetNodeId, param));
}

function findMaterialNode(targetMaterialComponent, targetType_, editorState) {
  var predNodeFunc = function (materialNode) {
    var match = MaterialNodeAssetService$WonderEditor.getNodeData(materialNode);
    if (match[/* materialComponent */1] === targetMaterialComponent) {
      return match[/* type_ */0] === targetType_;
    } else {
      return false;
    }
  };
  var arg = predNodeFunc;
  return (function (param) {
              return IterateTreeAssetService$WonderEditor.findOne(param, undefined, arg, undefined, undefined, /* () */0);
            })(TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
}

function insertNode(targetNodeId, newTreeNode, editorState) {
  return _getAndSetTree((function (param) {
                return OperateTreeAssetService$WonderEditor.insertNode(targetNodeId, newTreeNode, param);
              }), editorState);
}

function setNodeIsShowChildren(targetNodeId, isShowChildren, editorState) {
  return _getAndSetTree((function (param) {
                return OperateTreeAssetService$WonderEditor.setNodeIsShowChildren(targetNodeId, isShowChildren, param);
              }), editorState);
}

function removeNode(targetNode, editorState) {
  var newTree = OperateTreeAssetService$WonderEditor.removeNode(targetNode, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
  return TreeAssetEditorService$WonderEditor.setTree(newTree, editorState);
}

function removeNodeById(targetNodeId, editorState) {
  var match = OperateTreeAssetService$WonderEditor.removeNodeById(targetNodeId, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
  var newTree = match[0];
  return /* tuple */[
          (function (param) {
              return TreeAssetEditorService$WonderEditor.setTree(newTree, param);
            }),
          match[1]
        ];
}

function findNodeParent(targetNode, editorState) {
  return IterateTreeAssetService$WonderEditor.findOne(TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState), undefined, undefined, undefined, (function (node) {
                return Js_option.isSome(FolderNodeAssetService$WonderEditor.findChild(node, targetNode));
              }), /* () */0);
}

function findNodeParentId(targetNode, editorState) {
  return Js_option.map(NodeAssetService$WonderEditor.getNodeId, findNodeParent(targetNode, editorState));
}

function getCurrentNode(editorState) {
  return Js_option.map((function (nodeId) {
                var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
                return OperateTreeAssetService$WonderEditor.unsafeFindNodeById(nodeId, param);
              }), CurrentNodeIdAssetEditorService$WonderEditor.getCurrentNodeId(editorState));
}

function unsafeGetSelectedFolderNodeInAssetTree(editorState) {
  var targetNodeId = TreeAssetEditorService$WonderEditor.getSelectedFolderNodeIdInAssetTree(editorState);
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return OperateTreeAssetService$WonderEditor.unsafeFindNodeById(targetNodeId, param);
}

export {
  _getTree ,
  _getTreeWithLabel ,
  _getAndSetTree ,
  findNodeById ,
  unsafeFindNodeById ,
  isNodeExistById ,
  findMaterialNode ,
  insertNode ,
  setNodeIsShowChildren ,
  removeNode ,
  removeNodeById ,
  findNodeParent ,
  findNodeParentId ,
  getCurrentNode ,
  unsafeGetSelectedFolderNodeInAssetTree ,
  
}
/* FolderNodeAssetService-WonderEditor Not a pure module */
