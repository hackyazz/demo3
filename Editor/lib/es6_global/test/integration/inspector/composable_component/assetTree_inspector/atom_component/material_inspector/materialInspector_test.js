

import * as Curry from "../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as TestTool$WonderEditor from "../../../../../../tool/TestTool.js";
import * as ConsoleTool$WonderEditor from "../../../../../../unit/tool/external/ConsoleTool.js";
import * as InspectorTool$WonderEditor from "../../../../../../tool/ui/InspectorTool.js";
import * as ReactTestTool$WonderEditor from "../../../../../../tool/ReactTestTool.js";
import * as MaterialAssetTool$WonderEditor from "../../../../../asset/tool/MaterialAssetTool.js";
import * as BuildComponentTool$WonderEditor from "../../../../../../tool/BuildComponentTool.js";
import * as StateEditorService$WonderEditor from "../../../../../../../src/service/state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../../../../../../src/service/state/engine/StateEngineService.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../../tool/MainEditorSceneTool.js";
import * as AssetTreeInspectorTool$WonderEditor from "../../tool/AssetTreeInspectorTool.js";
import * as MainEditorMaterialTool$WonderEditor from "../../../sceneTree_inspector/renderGroup/material/tool/MainEditorMaterialTool.js";
import * as MaterialNodeAssetService$WonderEditor from "../../../../../../../src/service/record/editor/asset/MaterialNodeAssetService.js";
import * as NodeNameAssetLogicService$WonderEditor from "../../../../../../../src/service/stateTuple/logic/asset/NodeNameAssetLogicService.js";
import * as OperateTreeAssetEditorService$WonderEditor from "../../../../../../../src/service/state/editor/asset/OperateTreeAssetEditorService.js";
import * as MainEditorAssetChildrenNodeTool$WonderEditor from "../../../../../asset/tool/MainEditorAssetChildrenNodeTool.js";

describe("material inspector", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                return MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("test rename material", (function () {
                Wonder_jest.test("if type is basicMaterial, rename to default basic material name should not work", (function () {
                        ConsoleTool$WonderEditor.notShowMessage(/* () */0);
                        var addedMaterialNodeId = MaterialAssetTool$WonderEditor.addOneBasicMaterial(/* () */0);
                        var newName = MainEditorMaterialTool$WonderEditor.getDefaultBasicMaterialName(/* () */0);
                        AssetTreeInspectorTool$WonderEditor.Rename[/* renameAssetMaterialNode */2](undefined, undefined, addedMaterialNodeId, newName, /* () */0);
                        MainEditorAssetChildrenNodeTool$WonderEditor.selectMaterialNode(addedMaterialNodeId, undefined, undefined, /* () */0);
                        return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildInspectorComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), InspectorTool$WonderEditor.buildFakeAllShowComponentConfig(/* () */0)));
                      }));
                Wonder_jest.test("if type is lightMaterial, rename to default light material name should not work", (function () {
                        ConsoleTool$WonderEditor.notShowMessage(/* () */0);
                        var addedMaterialNodeId = MaterialAssetTool$WonderEditor.addOneLightMaterial(/* () */0);
                        var newName = MainEditorMaterialTool$WonderEditor.getDefaultLightMaterialName(/* () */0);
                        AssetTreeInspectorTool$WonderEditor.Rename[/* renameAssetMaterialNode */2](undefined, undefined, addedMaterialNodeId, newName, /* () */0);
                        MainEditorAssetChildrenNodeTool$WonderEditor.selectMaterialNode(addedMaterialNodeId, undefined, undefined, /* () */0);
                        return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildInspectorComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), InspectorTool$WonderEditor.buildFakeAllShowComponentConfig(/* () */0)));
                      }));
                describe("fix bug", (function () {
                        return Wonder_jest.test("if rename the same name, shouldn't warn", (function () {
                                      ConsoleTool$WonderEditor.notShowMessage(/* () */0);
                                      var warn = Sinon.createMethodStubWithJsObjSandbox(sandbox, console, "warn");
                                      var addedMaterialNodeId = MaterialAssetTool$WonderEditor.addOneLightMaterial(/* () */0);
                                      var editorState = StateEditorService$WonderEditor.getState(/* () */0);
                                      var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                      var match = MaterialNodeAssetService$WonderEditor.getNodeData(OperateTreeAssetEditorService$WonderEditor.unsafeFindNodeById(addedMaterialNodeId, editorState));
                                      var newName = NodeNameAssetLogicService$WonderEditor.getMaterialNodeName(match[/* materialComponent */1], match[/* type_ */0], engineState);
                                      AssetTreeInspectorTool$WonderEditor.Rename[/* renameAssetMaterialNode */2](undefined, undefined, addedMaterialNodeId, newName, /* () */0);
                                      return Sinon.toCalled(Wonder_jest.Expect[/* not_ */22](Wonder_jest.Expect[/* expect */0](warn)));
                                    }));
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
