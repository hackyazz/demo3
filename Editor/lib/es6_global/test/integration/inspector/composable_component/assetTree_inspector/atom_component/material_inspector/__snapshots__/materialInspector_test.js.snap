// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`material inspector test rename material if type is basicMaterial, rename to default basic material name should not work 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-material-inspector"
    >
      <h1>
        Material
      </h1>
      <hr />
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className="component-label"
          >
            Name
          </span>
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="New Material"
          />
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className=""
          >
            Type
          </span>
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="0"
            >
              basic_material
            </option>
            <option
              value="1"
            >
              light_material
            </option>
          </select>
        </div>
      </article>
      <article
        className="wonder-basic-material"
      >
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Color
          </div>
          <div
            className="item-content item-color"
          >
            <div
              className="color-hex"
              onClick={[Function]}
              style={
                Object {
                  "background": "#ffffff",
                }
              }
            />
            <div
              className="color-select"
              onClick={[Function]}
            >
              <img
                src="./public/img/color.png"
              />
            </div>
          </div>
        </article>
      </article>
    </article>
  </article>
</article>
`;

exports[`material inspector test rename material if type is lightMaterial, rename to default light material name should not work 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-material-inspector"
    >
      <h1>
        Material
      </h1>
      <hr />
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className="component-label"
          >
            Name
          </span>
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="New Material"
          />
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className=""
          >
            Type
          </span>
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="1"
          >
            <option
              value="0"
            >
              basic_material
            </option>
            <option
              value="1"
            >
              light_material
            </option>
          </select>
        </div>
      </article>
      <article
        className="wonder-light-material"
      >
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Diffcuse color
          </div>
          <div
            className="item-content item-color"
          >
            <div
              className="color-hex"
              onClick={[Function]}
              style={
                Object {
                  "background": "#ffffff",
                }
              }
            />
            <div
              className="color-select"
              onClick={[Function]}
            >
              <img
                src="./public/img/color.png"
              />
            </div>
          </div>
        </article>
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Diffuse map
          </div>
          <div
            className="item-content item-texture"
          >
            <div
              className="texture-img"
              onClick={[Function]}
              style={
                Object {
                  "opacity": "1",
                }
              }
            >
              <div
                className="img-dragBg"
                onDragEnter={[Function]}
                onDragLeave={[Function]}
                onDragOver={[Function]}
                onDrop={[Function]}
              />
            </div>
            <button
              className="texture-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </article>
        <article
          className="inspector-item wonder-float-input"
        >
          <div
            className="item-header component-label"
            onMouseDown={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
          >
            Shininess
          </div>
          <div
            className="item-content"
          >
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="32"
            />
          </div>
        </article>
      </article>
    </article>
  </article>
</article>
`;
